charts/internal/cloud-provider-config/templates/_cloud-provider-config-global.tpl

```diff
@@ -2,7 +2,13 @@
 auth-url="{{ .Values.authUrl }}"
 domain-name="{{ .Values.domainName }}"
 tenant-name="{{ .Values.tenantName }}"
+{{- if .Values.username }}
 username="{{ .Values.username }}"
 password="{{ .Values.password }}"
+{{- end }}
+{{- if .Values.applicationCredentialID }}
```

Comment:
I wanted to keep it simple. Especially if we want to support adding secrets in the Gardener dashboard, this can become quite cumbersome. It also produces additional test effort.
Is there any good reason to use `applicationCredentialName` (together with `userName` or a user id) instead of `applicationCredentialID`?

Reply:
> Is there any good reason to use...

`applicationCredentialName` may contain a hint for its purpose, but it adds additional complexity for providing a proper username and domain name, or just a userid.

Reply:
`applicationCredentialName` is supported now, too.

Reply:
@MartinWeindel I'd revert the `applicationCredentialName` support for now, because this may add extra confusion on the user end. See the detailed explanation of 3 auth cases: https://github.com/gophercloud/gophercloud/issues/1365#issuecomment-448294161

Reply:
P.S. I'm not against the `applicationCredentialName`, but in order to speed-up the appCreds implementation I'd focus on appID only. The appName support can be added later.

Reply:
As new releases are needed for both the Openstack provider and the machine-controller-provider-openstack, I don't think it will be faster to remove the appName support now as it has already been tested.
@dkistner WDYT?

Reply:
`applicationCredentialName` support has been removed again.